### PR TITLE
refactor(terraform-provider): refactor manager 

### PR DESCRIPTION
**Pull request by secustor**
This PR refactors the Terraform manager with the target to improve the code readability and minor bug fixes and improvements. 

The refactoring has been requested here: https://github.com/renovatebot/renovate/pull/6631#issuecomment-650872916

### CODE REVIEW

**Comment by secustor** [comment]
@rarkins Can you restart the pipeline?     
If I runt the testing suite locally there are no coverage issues.

**Comment by viceice** [comment]
@secustor see here:
image

https://codecov.io/gh/renovatebot/renovate/pull/6637/src/lib/manager/terraform/extract.ts?before=lib/manager/terraform/extract.ts

File: lib/manager/terraform/modules.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  });
+  return result;
+}
+
+export function analyseTerraformModule(dep: PackageDependency): void {
+  const githubRefMatch = githubRefMatchRegex.exec(dep.managerData.source);
+  const gitTagsRefMatch = gitTagsRefMatchRegex.exec(dep.managerData.source);
+  /* eslint-disable no-param-reassign */
+  if (githubRefMatch) {
+    const depNameShort = githubRefMatch.groups.project.replace(/\.git$/, '');
```
Comment (viceice) [inline lib/manager/terraform/modules.ts:34]: Please move regex to module scope `const`

File: lib/manager/terraform/modules.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    dep.lookupName = depNameShort;
+    if (!isVersion(dep.currentValue)) {
+      dep.skipReason = SkipReason.UnsupportedVersion;
+    }
+  } else if (gitTagsRefMatch) {
+    dep.depType = 'gitTags';
+    if (gitTagsRefMatch.groups.path.includes('//')) {
+      logger.debug('Terraform module contains subdirectory');
+      dep.depName = gitTagsRefMatch.groups.path.split('//')[0];
+      dep.depNameShort = dep.depName.split(/\/(.+)/)[1];
```
Comment (viceice) [inline lib/manager/terraform/modules.ts:49]: here too

**Comment by rarkins** [comment]
@secustor FYI codecov is a bit stricter about how it calculates coverage. @viceice has pasted from their output above

File: lib/manager/terraform/extract.ts [hunk 3]
```diff
@@ -58,128 +33,59 @@ export function extractPackageFile(content: string): PackageFile | null {
...
+    switch (dep.managerData.terraformDependencyType) {
+      case TerraformDependencyTypes.required_providers:
+      case TerraformDependencyTypes.provider:
+        analyzeTerraformProvider(dep);
+        break;
+      case TerraformDependencyTypes.module:
+        analyseTerraformModule(dep);
+        break;
+      /* istanbul ignore next */
+      default:
```
Comment (viceice) [inline lib/manager/terraform/extract.ts:85]: Do we really need the default?

**Review by viceice — COMMENTED** [review]
Just a small question otherwise LGTM

File: lib/manager/terraform/extract.ts (hunk 3 above)
Comment (secustor) [reply lib/manager/terraform/extract.ts:85]: This is a requirement of ES-lint, without it the linting fails.

**Comment by rarkins** [comment]
@secustor I received thousands of warning messages once I put this change live in the app, e.g.

image

FYI I have lowered this from WARN to TRACE because there seems to be nothing wrong here and it will otherwise spam the logs. If there are any cases you think should be caught and logged at a higher level, another PR would be welcome

**Comment by secustor** [comment]
Ah sry, I had this already reduced to trace in #6631, but it seems like it got lost during rebasing. 

I don't think there is a need to log this on a higher level.

**Pull request merged**